File: app/src/main/java/com/tommihirvonen/exifnotes/datastructures/FrameSortMode.java
package com.tommihirvonen.exifnotes.datastructures;

/**
 * Enum class to describe frame sort modes.
 */
public enum FrameSortMode {
    FRAME_COUNT(0),
    DATE(1),
    F_STOP(2),
    SHUTTER_SPEED(3),
    LENS(4);

    int value;

    FrameSortMode(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static FrameSortMode fromValue(int value) {
        switch (value) {
            case 0: default:
                return FRAME_COUNT;
            case 1:
                return DATE;
            case 2:
                return F_STOP;
            case 3:
                return SHUTTER_SPEED;
            case 4:
                return LENS;
        }
    }
}

File: app/src/main/java/com/tommihirvonen/exifnotes/datastructures/FrameSortMode.kt
package com.tommihirvonen.exifnotes.datastructures

enum class FrameSortMode constructor(value: Int) {
    FRAME_COUNT(0),
    DATE(1),
    F_STOP(2),
    SHUTTER_SPEED(3),
    LENS(4);

    var value: Int = 0
        internal set

    init {
        this.value = value
    }

    companion object {

        fun fromValue(value: Int): FrameSortMode {
            return when (value) {
                0 -> FRAME_COUNT
                1 -> DATE
                2 -> F_STOP
                3 -> SHUTTER_SPEED
                4 -> LENS
                else -> FRAME_COUNT
            }
        }
    }
}
